collect_app/src/main/java/org/odk/collect/android/formentry/QuitFormDialogFragment.java

```diff
@@ -0,0 +1,135 @@
+package org.odk.collect.android.formentry;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.widget.ListView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.google.common.collect.ImmutableList;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.adapters.IconMenuListAdapter;
+import org.odk.collect.android.adapters.model.IconMenuItem;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.dao.helpers.InstancesDaoHelper;
+import org.odk.collect.android.external.ExternalDataManager;
+import org.odk.collect.android.formentry.audit.AuditEvent;
+import org.odk.collect.android.formentry.saving.FormSaveViewModel;
+import org.odk.collect.android.preferences.AdminKeys;
+import org.odk.collect.android.preferences.AdminSharedPreferences;
+import org.odk.collect.android.utilities.DialogUtils;
+import org.odk.collect.android.utilities.MediaManager;
+
+import java.util.List;
+
+import static android.app.Activity.RESULT_OK;
+
+public class QuitFormDialogFragment extends DialogFragment {
+
+    private FormSaveViewModel viewModel;
+    private final ViewModelProvider.Factory viewModelFactory = new FormSaveViewModel.Factory();
+
+    private ListView listView;
+    private String title;
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        viewModel = ViewModelProviders.of(requireActivity(), viewModelFactory).get(FormSaveViewModel.class);
+
+        title = viewModel.getFormName();
```

Comment:
All this view initialization can probably go in `onCreateDialog` instead of here so you don't need the `title` and `listView` fields.